app/services/proxy_status/base.rb

```diff
@@ -0,0 +1,43 @@
+module ProxyStatus
+  class Base
+    def initialize(proxy, opts = {})
+      @proxy = proxy
+      @cache_duration = opts[:cache_duration] || 3.minutes
+      @cache_status = opts[:cache_status].nil? ? true : opts[:cache_status]
+    end
+
+    def revoke_cache!
+      Rails.cache.delete(cache_key)
+    end
+
+    def cache_key
+      raise NotImplementedError, _("cache_key not implemented for ProxyStatus::#{self.class}")
+    end
+
+    def self.humanized_name
+      # Should be named as the proxy feature name!
+      'Base'
+    end
+
+    private
+
+    attr_reader :proxy, :cache_duration, :api_class_name, :cache_status
+
+    def api
+      @api = "ProxyAPI::#{self.class.humanized_name}".classify.constantize.new(:url => proxy.url)
```

Comment:
what happens if no such class exists? afaiu this will be reported in the UI as unable to initialize class name or something like that.
I wonder if we should have a nicer error or its fine as its developer oriented?